internal/gatewayapi/resource.go

```diff
@@ -20,6 +20,37 @@ import (
 
 type XdsIRMap map[string]*ir.Xds
 type InfraIRMap map[string]*ir.Infra
+type GatewayClassResources map[string]*Resources
+
+// DeepCopyInto is an autogenerated deepcopy function, copying the receiver, writing into out. in must be non-nil.
```

Comment:
did you hand write this ?
you can use a annotation similar to https://github.com/envoyproxy/gateway/blob/988d4ed0be5a5f9dd39a784d52c3bb5c51ecc2aa/internal/gatewayapi/resource.go#L26 to autogenerate it

Reply:
I did generate it, I will add comment
deepcopy-gen returns GatewayClassResources, I needed to change it to return *GatewayClassResources

Reply:
because its a `map` ? , if it was a list would we hit this same issue ?

Reply:
yes, I think it is because of the map